tests/network/hotplug.go

```diff
@@ -44,174 +44,233 @@ import (
 	"kubevirt.io/kubevirt/tests/libnet"
 	"kubevirt.io/kubevirt/tests/libvmi"
 	"kubevirt.io/kubevirt/tests/libwait"
+	"kubevirt.io/kubevirt/tests/testsuite"
 	"kubevirt.io/kubevirt/tests/util"
 )
 
-var _ = SIGDescribe("nic-hotplug", decorators.InPlaceHotplugNICs, func() {
+const (
+	linuxBridgeName = "supadupabr"
+	ifaceName       = "iface1"
+	networkName     = "skynet"
+	vmIfaceName     = "eth1"
+)
+
+type hotplugMethod string
+
+const (
+	migrationBased hotplugMethod = "migrationBased"
+	inPlace        hotplugMethod = "inPlace"
+)
+
+var _ = SIGDescribe("nic-hotplug", func() {
+	verifyHotplug := func(vmi *v1.VirtualMachineInstance, plugMethod hotplugMethod) *v1.VirtualMachineInstance {
```

Comment:
**nit**: I would set this function to be regular function and not anonymous (as it was before)

Reply:
I find it ugly to have some of the methods declared before the sig-describe and some after.

Reply:
Please note that in general, such functions may introduce side effects if they use a variable that was defined earlier in an implicit way.
As a reviewer, I immediately suspect that it mutates a variable outside the function scope.
for example: 
```
var myTestVMI *v1.VirtualMachineInstance
...
waitVMI := func() {
    myTestVMI =  ...
}
...
```

Still a nit though.